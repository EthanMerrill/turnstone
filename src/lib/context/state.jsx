import React, { createContext, useReducer, useState, useEffect } from 'react'
import reducer from '../reducers/reducer'
import {setQuery} from '../actions/actions'
import undef from '../utils/undef'

const StateContext = createContext() //TODO: Rename GlobalStateContext

const StateContextProvider = (props) => {
  const { children, splitChar, styles = {}, text = '', items = [] } = props
  const [state, dispatch] = useReducer(reducer, {
    query: text,
    items,
    highlighted: undef,
    selected: undef,
    customStyles: styles,
    splitChar
  })
  const [selectedState, setSelectedState] = useState()
  const [customStyles] = useState(styles)
  const [splitCharState] = useState(splitChar)

  useEffect(() => dispatch(setQuery(text)), [text]) // TODO: Is this needed?

  return (
    <StateContext.Provider
      value={{
        state,
        dispatch,
        selectedState,
        setSelectedState,
        customStyles,
        splitCharState
      }}>
      {children}
    </StateContext.Provider>
  )
}

export { StateContext, StateContextProvider }
