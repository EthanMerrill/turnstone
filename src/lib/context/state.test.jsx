import React from 'react'
import renderer from 'react-test-renderer'

import { describe, expect, test } from 'vitest'
import { StateContextProvider, StateContext } from './state.jsx'

describe('StateContextProvider', () => {
  test('StateContextProvider component renders', () => {
    const component = renderer.create(
      <StateContextProvider>Test</StateContextProvider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('StateContext', () => {
  test('StateContext is present', () => {
    expect(StateContext && typeof StateContext === 'object').toBe(true)
  })
})