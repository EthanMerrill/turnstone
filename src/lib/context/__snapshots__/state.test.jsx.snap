// Vitest Snapshot v1

exports[`StateContextProvider > StateContextProvider component renders 1`] = `"Test"`;
